File: reflexion/Ejemplos.Reflection/Ejemplos basicos/Test_Reflection/Persona.vb
Imports System

Public Class Persona

    Private _nombre As String
    Private _apellido As String

    ''' <summary>
    ''' 
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property Nombre() As String
        Get
            Return _nombre
        End Get
        Set(ByVal value As String)
            _nombre = value
        End Set
    End Property

    ''' <summary>
    ''' 
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Property Apellido() As String
        Get
            Return _apellido
        End Get
        Set(ByVal value As String)
            _apellido = value
        End Set
    End Property

    ''' <summary>
    ''' 
    ''' </summary>
    ''' <remarks></remarks>
    Public Sub Saludar()
        Console.WriteLine("Test_Reflection.dll::Public Sub Saludar()")
    End Sub

    Public Sub Saludar(ByVal valor As String)
        Console.WriteLine("Test_Reflection.dll::Public Sub Saludar() " + valor)
    End Sub

    Private Sub SaludarDos()
        Console.WriteLine("Test_Reflection.dll::Private Sub SaludarDos()")
    End Sub


End Class

File: reflexion/Ejemplos.Reflection/Reflection C#/Persona.cs
namespace Test_Reflection
{
    using System;

    public class Persona
    {
        private string _apellido;
        private string _nombre;

        //private Persona()
        //{

        //}

        public Persona()
        {
                
        }
        //public Persona(string apellido, string nombre)
        //{
        //    this._apellido = apellido;
        //    this._nombre = nombre;
        //}

        public void Saludar()
        {
            Console.WriteLine("Test_Reflection.dll::Public Sub Saludar()");
        }

        public void Saludar(string valor)
        {
            Console.WriteLine("Test_Reflection.dll::Public Sub Saludar() " + valor);
        }

        private void SaludarDos()
        {
            Console.WriteLine("Test_Reflection.dll::Private Sub SaludarDos()");
        }

        public string Apellido
        {
            get
            {
                return this._apellido;
            }
            set
            {
                this._apellido = value;
            }
        }

        public string Nombre
        {
            get
            {
                return this._nombre;
            }
            set
            {
                this._nombre = value;
            }
        }
    }
}

